packages/devtools_app/lib/src/screens/performance/panes/timeline_events/timeline_events_controller.dart

```diff
@@ -327,6 +325,33 @@ class TimelineEventsController extends PerformanceFeatureController
     });
   }
 
+  Future<void> _perfettoSelectFrame(FlutterFrame frame) async {
+    if (!offlineController.offlineMode.value) {
```

Comment:
Would early return make it more readable?

if (offlineController.offlineMode.value) return;

Reply:
no because there is a call to `perfettoController.scrollToTimeRange` at the end of the method that needs to be hit regardless of whether we are in offline mode or not